File: typescript/base/src/main/java/lang/Character.ts
export class Character {
  private static toCodePoint(value: number | string): number {
    if (typeof value === "number") {
      return value;
    }
    if (!value || value.length === 0) {
      return 0;
    }
    return value.charCodeAt(0);
  }

  public static isDigit(value: number | string): boolean {
    const code = Character.toCodePoint(value);
    return code >= 48 && code <= 57;
  }

  public static isSpace(value: number | string): boolean {
    const code = Character.toCodePoint(value);
    return code === 32 || (code >= 9 && code <= 13);
  }

  public static isLetter(value: number | string): boolean {
    const code = Character.toCodePoint(value);
    return (code >= 65 && code <= 90) || (code >= 97 && code <= 122);
  }
}

File: cpp/base/src/main/java/lang/Character.cpp
#include "java/lang/Character.h"

namespace java {

bool
Character::isDigit(int value) {
    return value >= '0' && value <= '9';
}

bool
Character::isSpace(int value) {
    return value == ' ' || (value >= 9 && value <= 13);
}

bool
Character::isLetter(int value) {
    return (value >= 'A' && value <= 'Z')
           || (value >= 'a' && value <= 'z');
}

}
